lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -97,7 +97,7 @@ final class PersistentCollection extends AbstractLazyCollection implements Selec
      *
      * @param EntityManagerInterface $em    The EntityManager the collection will be associated with.
      * @param ClassMetadata          $class The class descriptor of the entity type of this collection.
-     * @psalm-param Collection<TKey, T> $collection The collection elements.
+     * @psalm-param Collection<TKey, T>&Selectable<TKey, T> $collection The collection elements.
      */
     public function __construct(EntityManagerInterface $em, $class, Collection $collection)
```

Comment:
Since `unwrap` need to return a Selectable (cf next comment), we need to only accepts selectable.